src/lib/API/api.js

```diff
@@ -475,7 +476,13 @@ export class APIService {
       baseURL: explorer,
     })
 
-    return result?.ContractName
+    const res = Array.isArray(result) ? result[0] : result
+
+    if (res?.Proxy === '1') {
+      return this.getContractName(res.Implementation, chainId, explorer)
```

Comment:
have you verified the var name `Implementation` is available via all explorer APIs? both those based on blockscout (explorer.celo.org/explorer.fuse.io) and those based on etherscan. specifically the uppercase `I`?

Reply:
No I hadn't, is fixed now